core/datasource/internal/timetrigger/timetrigger.go

```diff
@@ -31,6 +31,15 @@ type SpecConfiguration struct {
 	Conditions []*common.SpecCondition
 }
 
+func (s SpecConfiguration) SetInitial(initial, timeNow time.Time) error {
```

Comment:
A pointer receiver would make it more clear that this call alters `SpecConfiguration`, I also think the name `SetInitial` is a bit confusing, and `SetFirst` or something makes more sense

Reply:
yea well the field was name initial... it's a setter... so eh